skafka/src/main/scala/com/evolutiongaming/skafka/consumer/RebalanceCallback.scala

```diff
@@ -0,0 +1,267 @@
+package com.evolutiongaming.skafka.consumer
+
+import java.lang.{Long => LongJ}
+import java.time.Instant
+import java.util.{Map => MapJ}
+
+import cats.data.{NonEmptyMap => Nem, NonEmptySet => Nes}
+import cats.implicits._
+import cats.{Monad, ~>}
+import com.evolutiongaming.catshelper.CatsHelper._
+import com.evolutiongaming.catshelper.{ApplicativeThrowable, MonadThrowable, ToTry}
+import com.evolutiongaming.skafka.Converters._
+import com.evolutiongaming.skafka._
+import com.evolutiongaming.skafka.consumer.ConsumerConverters._
+import com.evolutiongaming.skafka.consumer.RebalanceCallback.Helpers._
+import org.apache.kafka.clients.consumer.{Consumer => ConsumerJ}
+import org.apache.kafka.common.{TopicPartition => TopicPartitionJ}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+// TODO review if we can do it without F[_] everywhere
+// TODO add most common usages in scaladoc from spec (using position/seek/commit)
+/*
+def onPartitionAssigned(callback: RebalanceCallback[Unit]): ListenerConsumer[Unit] = {
+  for {
+    position <- consumer.position
+    saved <- ListenerConsumer.lift { db.save(…) }
+    result <- {
+        if (saved) consumer.commit(…)
+        else ListenerConsumer.unit
+      }
+  } yield result
+}
+ */
+/**
+  * Allows to describe computations in callback methods of [[org.apache.kafka.clients.consumer.ConsumerRebalanceListener]]
+  */
+sealed trait RebalanceCallback[F[_], A] {
+  def mapK[G[_]](fg: F ~> G): RebalanceCallback[G, A]
+}
+
+object RebalanceCallback {
```

Comment:
could you please add link to ticket about creating implementation of 

```
implicit def monadRebalanceCallback[F[_]]: Monad[RebalanceCallback[F, *]] = new Monad[RebalanceCallback[F, *]] {…}
```

Reply:
added Monad instance in scope of latest commits